docs/utils-reference/react-hooks/useStreamJSON.md

```diff
@@ -0,0 +1,217 @@
+# `useStreamJSON`
+
+Hook which takes a `http://`, `https://` or `file:///` URL pointing to a JSON resource, caches it to the command's support folder, and streams through its content. Useful when dealing with large JSON arrays which would be too big to fit in the command's memory.
+
+## Signature
+
+```ts
+export function useStreamJSON<T, U>(
+  url: RequestInfo,
+  options: RequestInit & {
+    fileName?: string;
+    folder?: string;
```

Comment:
I'm a bit confused about those 2 options. Do we need them? Can't we use a hash of the url as the filename?

Reply:
they're useful if you want to use the hook twice for the same data source – however, since we added `dataPath` and you can stream items from multiple arrays in the same data source, it shouldn't be as useful anymore 🤔

We can remove them and see if they get requested 😸

Reply:
The same data source should have the same URL, no?

Reply:
yup 😅